### PR TITLE
feature(frontend): switch language client side

### PR DESCRIPTION
**Pull request by sebastien-comeau**

### CODE REVIEW

File: frontend/app/routes/api.switch-language.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  language: z.enum(['en', 'fr']),
+});
+
+export type SwitchLanguageData = z.infer<typeof switchLanguageDataSchema>;
+
+/**
+ * A loader function that always throws a 404 Not Found response.
+ */
+export async function loader() {
+  throw json('404 Not Found', { status: 404 });
```
Comment (gregory-j-baker) [inline frontend/app/routes/api.switch-language.ts:20]: Suggestion: return a 405 status. <https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405>

File: frontend/app/routes/api.switch-language.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * @example
+ * const request = //... create a request object
+ * const response = await action({ request });
+ * console.log(response.status); // Output: 204
+ *
+ * @param {ActionFunctionArgs} args - The action function arguments.
+ */
+export async function action({ request }: ActionFunctionArgs) {
+  if (request.method !== 'POST') {
+    log.debug(`Status 405: Invalid request method ${request.method}`);
```
Comment (gregory-j-baker) [inline frontend/app/routes/api.switch-language.ts:38]: Should this be logged as a warning, maybe?

File: frontend/app/routes/api.switch-language.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    throw json(`405 Invalid request method ${request.method}`, { status: 405 });
+  }
+
+  if (request.headers.get('Content-Type') !== 'application/json') {
+    log.debug(`Status 415: Invalid media type ${request.headers.get('Content-Type')}`);
```
Comment (gregory-j-baker) [inline frontend/app/routes/api.switch-language.ts:43]: Should this also be logged as a warning?

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]

File: frontend/app/routes/api.switch-language.ts (hunk 1 above)
Comment (sebastien-comeau) [reply frontend/app/routes/api.switch-language.ts]: Fixed

File: frontend/app/routes/api.switch-language.ts (hunk 2 above)
Comment (sebastien-comeau) [reply frontend/app/routes/api.switch-language.ts]: Fixed

File: frontend/app/routes/api.switch-language.ts (hunk 3 above)
Comment (sebastien-comeau) [reply frontend/app/routes/api.switch-language.ts]: Fixed

**Review by gregory-j-baker — APPROVED** [review]

**Pull request merged**